src/test/java/com/google/javascript/gents/singleTests/inline_types.js

```diff
@@ -5,6 +5,6 @@ var /** !Foo */ foo = {};
 
 const /** string */ x = 'fruit';
 
-// TODO(#352): Support inline typing for functions.
-// input: function /** string */ f(/** number */ x) {return x + ' apples'}
-// output: function f(x: number): string {return x + ' apples'}
+function /** string */ f(/** number */ x, z, /** string */ y) {
```

Comment:
Thank you for a good idea. Adding a second version uncovered a flaw in my solution. I pushed a fix.